File: Difficulty: Easy/Second Largest/second-largest.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.print2largest(arr);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int print2largest(int[] arr) {
        int max = arr[0];
        int sMax = -1;
        
        for(int i: arr){
            if(i<max){
                sMax = Math.max(i, sMax);
            } else if(i>max){
                sMax = max;
                max = i;
            }
        }
        return sMax;
    }
}

File: School/Second Largest/second-largest.py
#User function Template for python3
class Solution:
    def print2largest(self, arr, n):
        l = arr[0]
        sl = -1
        for i in arr:
            if i < l:
                sl = max(sl, i)
            elif i > l:
                sl = l
                l = i
        return sl


#{ 
 # Driver Code Starts
#Initial Template for Python 3



if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.print2largest(arr, n)
        print(ans)
        tc -= 1

# } Driver Code Ends